src/main/java/io/vertx/core/http/impl/HttpServerChannelInitializer.java

```diff
@@ -217,6 +218,7 @@ private void configureHttp1OrH2C(ChannelPipeline pipeline, HandlerHolder<? exten
     if (HttpServerImpl.USE_FLASH_POLICY_HANDLER) {
       pipeline.addLast("flashpolicy", new FlashPolicyHandler());
     }
+    pipeline.addLast("optionalHAProxyMessageDecoder", new OptionalHAProxyMessageDecoder());
```

Comment:
we should only add this to the pipeline when the option is configured. pipeline handlers have effect on performance and we should avoid adding them when it is not necessary

Reply:
so there should be an option for enabling / disabling it in server

Reply:
also this will allow to run without the ha proxy class on the classpath by default

Reply:
Have in mind that its autoremoved once the first 12 bytes are received (12 bytes are the number of bytes required to find out if its an haproxy protocol)

Instead of an option we could check if HaProxy is in classpatch and enable it

Tell me which one you prefer and will implement

Option or classpath?

Reply:
good point for the removal, for now let's do classpath as I would like to avoid having the jar OOTB.